sdk/search/Azure.Search.Documents/src/Indexes/Models/IndexingParameters.cs

```diff
@@ -1,20 +1,416 @@
-        /// sample for code you can include in your own projects to set these values intuitively.
+        /// Use <see cref="IndexingParametersConfiguration"/> instead to set well-known properties intuitively.
         /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public IDictionary<string, object> Configuration { get; }
+
+        /// <summary>
+        /// Indexer-specific configuration properties.
+        /// Each value must be of a primitive type.
+        /// </summary>
+        [CodeGenMember("Configuration")]
+        public IndexingParametersConfiguration IndexingParametersConfiguration
+        {
+            get => _indexingParametersConfiguration;
+            set => _indexingParametersConfiguration = value;
+        }
+
+        private class ConfigurationAdapter : IDictionary<string, object>
+        {
+            private readonly IndexingParameters _parameters;
+
+            public ConfigurationAdapter(IndexingParameters parameters)
+            {
+                _parameters = parameters ?? throw new ArgumentNullException(nameof(parameters));
+
+                Keys = new ConfigurationAdapterKeysCollection(this);
+                Values = new ConfigurationAdapterValuesCollection(this);
+            }
+
+            /// <inheritdoc />
+            public object this[string key]
+            {
+                get => TryGetValue(key, out object value) ? value : throw new KeyNotFoundException($"{key} not found");
+                set
+                {
+                    EnsureInitialized();
+
+                    IndexingParametersConfiguration configuration = _parameters._indexingParametersConfiguration;
+
+                    if (IndexingParametersConfiguration.WellKnownProperties.TryGetValue(key, out (Func<IndexingParametersConfiguration, object> Get, Action<IndexingParametersConfiguration, object> Set) property))
+                    {
+                        property.Set(configuration, value);
+                    }
+                    else
+                    {
+                        configuration[key] = value;
+                    }
+                }
+            }
+
+            /// <inheritdoc />
+            public ICollection<string> Keys { get; }
+
+            /// <inheritdoc />
+            public ICollection<object> Values { get; }
+
+            /// <inheritdoc />
+            public int Count
+            {
+                get
+                {
+                    IndexingParametersConfiguration configuration = _parameters._indexingParametersConfiguration;
+                    if (configuration is null)
+                    {
+                        return 0;
+                    }
+
+                    // Have to count ourselves since Enumerable.Count() will call into this property causing a StackOverflowException.
```

Comment:
Whoops :)

Reply:
The concept was there. I had to make a few changes anyway, but your founding idea with the dictionary of delegates was great! I got the dictionary I wanted to handle most of the abstraction, and without maintaining separate dictionaries as before to aggregate multiple other dictionaries. This is much cleaner. 😉